sdk/aqueduct/decorator.py

```diff
@@ -208,21 +208,42 @@ def _type_check_decorated_function_arguments(
                 )
 
 
-def _convert_argument_to_parameter(
-    *input_artifacts: Any, function_argument_names: List[str]
+def _convert_input_arguments_to_parameters(
+    *input_artifacts: Any, func_params: Mapping[str, inspect.Parameter]
 ) -> List[BaseArtifact]:
     """
     Converts non-artifact inputs to parameters.
+
+    Errors if the implicitly-created parameter collides with an existing one. Also errors if the function has a
+    variable-length parameter, since we don't know what name to attribute to those.
+
+    `func_params` maps from parameter name to an `inspect.Parameter` object containing additional information.
     """
+    # KEYWORD_ONLY parameters are allowed, since they are guaranteed to have a name.
+    # Note that we only accept them after "*" arguments, since we error out on VAR_POSITIONAL (eg. *args).
+    disallowed_kinds = [inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD]
```

Comment:
I'm also a bit confused by what each of these option means. Maybe at least giving a link to the documentation that lists all options would be helpful.